### PR TITLE
JAMES-2744 James should provide Content-Type when downloading

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[b7fa04f138e41fced2bf907aa9b1261fa682285d] OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x0000000625c00000, 6882852864, 0) failed; error='Cannot allocate memory' (errno=12)
[b7fa04f138e41fced2bf907aa9b1261fa682285d] 
[b7fa04f138e41fced2bf907aa9b1261fa682285d] #
[b7fa04f138e41fced2bf907aa9b1261fa682285d] 
[b7fa04f138e41fced2bf907aa9b1261fa682285d] # There is insufficient memory for the Java Runtime Environment to continue.
[b7fa04f138e41fced2bf907aa9b1261fa682285d] 
[b7fa04f138e41fced2bf907aa9b1261fa682285d] # Native memory allocation (mmap) failed to map 6882852864 bytes for committing reserved memory.
[b7fa04f138e41fced2bf907aa9b1261fa682285d] 
[b7fa04f138e41fced2bf907aa9b1261fa682285d] # An error report file with more information is saved as:
[b7fa04f138e41fced2bf907aa9b1261fa682285d] 
[b7fa04f138e41fced2bf907aa9b1261fa682285d] # /tmp/hs_err_pid1.log
```
test this please

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[b7fa04f138e41fced2bf907aa9b1261fa682285d] Switched to a new branch 'master'
[b7fa04f138e41fced2bf907aa9b1261fa682285d] fatal: Cannot update paths and switch to branch 'b7fa04f138e41fced2bf907aa9b1261fa682285d' at the same time.
```
what the hell. test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 1]
```diff
@@ -69,6 +69,12 @@ Feature: Download GET
     Then she can read that blob
     And the blob size is 36
 
+  Scenario: Getting a message then getting its Content-Type of the blob
+    Given "alice@domain.tld" has a message "m1" in "INBOX" mailbox with subject "my test subject", content "testmail"
+    And "alice@domain.tld" ask for messages "m1"
+    When "alice@domain.tld" downloads should not return Content-Type of the message by its blobId
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: This sentence doesn't make sense: it mixes "When" (the action phase") with an assertion "should not return Content-type".
Please split it in two.
Btw, we don't care to keep the last line.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 2]
```diff
@@ -72,7 +72,7 @@ Feature: Download GET
   Scenario: Getting a message then getting its Content-Type of the blob
     Given "alice@domain.tld" has a message "m1" in "INBOX" mailbox with subject "my test subject", content "testmail"
     And "alice@domain.tld" ask for messages "m1"
-    When "alice@domain.tld" downloads should not return Content-Type of the message by its blobId
+    When "alice@domain.tld" downloads should return Content-Type of the message by its blobId
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: the test must be more precise: it should enforce the expected content-type

**Comment by aduprat** [comment]
:green_apple: before CI restart

**Comment by trantienduchn** [comment]
having a scenario of  appending a message with an attachement and a dynamic content type is quite hard for me to implement

**Comment by mbaechler** [comment]
> having a scenario of appending a message with an attachement and a dynamic content type is quite hard for me to implement

Really? Because the current gatling scenario doesn't make sense: we talk about an attachment and its type appear only in the assertion. There's no way this test would work without an arbitrary implementation of the steps, which is a sign that the gatling spec is not ok.

File: server/container/util/src/test/java/org/apache/james/util/InputStreamUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.google.common.primitives.Bytes;
+
+class InputStreamUtilsTest {
+    private static final byte[] CONTENT_1 = "content1".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] CONTENT_2 = "content2".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] EMPTY = "".getBytes(StandardCharsets.UTF_8);
+
+    @Test
+    void concatShouldNoopWhenSingleAgument() {
```
Comment (rouazana) [inline server/container/util/src/test/java/org/apache/james/util/InputStreamUtilsTest.java]: s/Agument/Argument

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**